src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Unix.cs

```diff
@@ -10,47 +10,77 @@
 using System.Runtime.InteropServices;
 using System.Threading;
 using Microsoft.DotNet.XUnitExtensions;
+using Microsoft.DotNet.RemoteExecutor;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace System.IO.Tests
 {
-    public partial class DangerousFileSystemWatcherTests
+    // Helper class to avoid issues with RemoteExecutor and _output and other properties.
```

Comment:
What issues and how is this working around them?  I'm not seeing it.

Reply:
I'm getting 
```
     System.IO.Tests.DangerousFileSystemWatcherTests.FileSystemWatcher_Unix_DoesNotLeak [FAIL]
        Assert.All() Failure: 1 out of 1 items in the collection did not pass.
        [0]: Item: Xunit.Abstractions.ITestOutputHelper _output
             Xunit.Sdk.TrueException: Field marshaling is not supported by Invoke: _output
             Expected: True
             Actual:   False
                at Xunit.Assert.True(Nullable`1 condition, String userMessage) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\BooleanAsserts.cs:line 96
                at Xunit.Assert.True(Boolean condition, String userMessage) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\BooleanAsserts.cs:line 84
                at Microsoft.DotNet.RemoteExecutor.RemoteExecutor.<>c.<GetMethodInfo>b__24_0(FieldInfo fi) in /_/src/Microsoft.DotNet.RemoteExecutor/src/RemoteExecutor.cs:line 351
                at Xunit.Assert.All[T](IEnumerable`1 collection, Action`1 action) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\CollectionAsserts.cs:line 36
        Stack Trace:
          /_/src/Microsoft.DotNet.RemoteExecutor/src/RemoteExecutor.cs(354,0): at Microsoft.DotNet.RemoteExecutor.RemoteExecutor.GetMethodInfo(Delegate d)
          /_/src/Microsoft.DotNet.RemoteExecutor/src/RemoteExecutor.cs(212,0): at Microsoft.DotNet.RemoteExecutor.RemoteExecutor.Invoke(Func`2 method, String arg, RemoteInvokeOptions options)
          /Users/furt/github/wfurt-runtime/src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Unix.cs(89,0): at System.IO.Tests.DangerousFileSystemWatcherTests.FileSystemWatcher_Unix_DoesNotLeak()
```

and `_output` is _not_  used inside of the remote code. I had a prototype where I ditched _outout completely and it worked but I somewhat like the ability to instrument. 

I don't know if there is any better way how to get around this.

Reply:
You're getting that with what code?

Reply:
when I move the body of `ExecuteTest()` to the `FileSystemWatcher_Unix_DoesNotLeak()`  itself. For that reason, I moved the test body to separate class to defeat checks from `GetMethodInfo()`

Reply:
I the issue and it is working now without extra class.